File: BubbleSort/BubbleSort.c
#include <stdio.h>
#include <malloc.h>

int main(){
    int i, n, changes, _changes = 0; 

    scanf("%d",&n);
    
    int *a = malloc(sizeof(int) * n);
    
    for(i = 0; i < n; i++){

       scanf("%d", &a[i]);
    }
    
    do {
        
       changes = 0;
        
       for(i = 0; i < n - 1; i++)
           
           if(a[i] > a[i + 1]) {
               
               int x = a[i]^a[i+1];
                   a[i] = x^a[i];
                   a[i+1] = x^a[i+1];
                    changes++;
                   _changes++;
           }
                              
    } while(changes);
    
    printf("Array is sorted in %d swaps.\nFirst Element: %d\nLast Element: %d", _changes, a[0], a[n-1]);    
    return 0;
}

File: BubbleSort/BubbleSort.go
package main
import "fmt"

func main() {
    var n int
    fmt.Scanf("%d", &n)
    a := make([]int, n)
    for i := 0; i < n; i++ {
        fmt.Scanf("%d", &a[i])
    }
    var swaps int
    for i := 0; i < n; i++ {
        var x int
        for j := 0; j < n-1; j++ {
            if a[j] > a[j+1] {
                a[j], a[j+1] = a[j+1], a[j]
                x++
            }
        }
        if x == 0 {
            break
        }
        swaps += x
    }
    fmt.Println("Array is sorted in", swaps, "swaps.")
    fmt.Println("First Element:", a[0])
    fmt.Println("Last Element:", a[n-1])